src/CreateEntityMappingView/CreateEntityMappingView.js

```diff
@@ -23,64 +24,121 @@ class CreateEntityMappingView extends ConnectionComponent {
+		this.handleTargetStatusChange = this.handleTargetStatusChange.bind(this);
+		this.prepareStateForSubmitting = this.prepareStateForSubmitting.bind(this);
+		this.submitNewMapping = this.submitNewMapping.bind(this);
+		this.submitUpdatedMapping = this.submitUpdatedMapping.bind(this);
+		this.entityMappingPending = this.entityMappingPending.bind(this);
+		this.fetchInitialData = this.fetchInitialData.bind(this);
+		this.renderComponentBody = this.renderComponentBody.bind(this);
+
 	}
 
+	// returns the default mappings, called on load of create view and event or event type change
 	static getDefaultMappings() {
 		return [{
 			eventTypeAttribute: {value: null, errorMessage: ""},
 			entityTypeAttribute: {value: null, errorMessage: ""}
 		}];
 	}
 
-	static getMenuItems(items) {
-		return items.map(
-			(item, key) => {
-				return <MenuItem
-					key={key}
-					value={item.Id}
-					primaryText={item.Name}/>;
+	// returns the state loaded from a response
+	static getLoadedStateFromResponse(entityMapping) {
+		let mappings = [];
+		entityMapping.EventEntityMappingConditions.forEach(mappingCondition => {
+			mappings.push({
+				eventTypeAttribute: {value: mappingCondition.EventTypeAttributeId, errorMessage: ""},
+				entityTypeAttribute: {value: mappingCondition.EntityTypeAttributeId, errorMessage: ""}
 			});
+		});
+		return {
+			targetStatus: entityMapping.TargetStatus,
+			selectedEventTypeId: {value: entityMapping.EventTypeId},
+			selectedEntityTypeId: {value: entityMapping.EntityTypeId},
+			mappings: mappings
+		};
 	}
 
-    componentDidUpdate(prevProps, prevState) {
-        if(prevState.selectedEventTypeId.value !== this.state.selectedEventTypeId.value) {
-            this.props.lazyEventTypeAttributeLoading(this.state.selectedEventTypeId.value);
-        }
-        if(prevState.selectedEntityTypeId.value !== this.state.selectedEntityTypeId.value) {
-            this.props.lazyEntityTypeAttributeLoading(this.state.selectedEntityTypeId.value);
-        }
-    }
-    
-    submitMapping() {
-		if(this.isValidInput()) {
-			const  entityMappingConditions = this.state.mappings.map((mappingStatement) => {
-				return ({
-					EventTypeAttributeId: mappingStatement.eventTypeAttribute.value,
-					EntityTypeAttributeId: mappingStatement.entityTypeAttribute.value
-				});
+	isValidInput() {
+		let isValid = true;
+		if(this.state.selectedEntityTypeId.value === null) {
+			this.setState({
+				selectedEntityTypeId: {value: null, errorMessage: config.messages.requiredFieldMessage}
 			});
-			this.props.createEntityMapping({
-				EventTypeId: this.state.selectedEventTypeId.value,
-				EntityTypeId: this.state.selectedEntityTypeId.value,
-				TargetStatus: this.state.targetStatus,
-				EventEntityMappingConditions: entityMappingConditions
+			isValid = false;
+		}
+		if (this.state.selectedEventTypeId.value === null) {
+			this.setState({
+				selectedEventTypeId: {value: null, errorMessage: config.messages.requiredFieldMessage}
```

Comment:
`selectedEventTypeId` seems to be an object and not an int? Maybe the naming is not appropriate.